source/common.ts

```diff
@@ -0,0 +1,46 @@
+import { createOwnCloudFileDetailProperty } from './helper';
+import { NextcloudClientInterface }         from './types';
+import { dirname, basename }                from 'path';
+import * as assert                          from 'assert';
+
+export async function getFileOrFolderCreator(fullPath) : Promise<string> {
+  const self: NextcloudClientInterface = this;
+
+  let result = null;
+
+  const baseFolder       = dirname(fullPath);
+  const fileOrFolderName = basename(fullPath);
+
+  try {
+    assert(await self.exists(baseFolder));
+    const folderFileDetails = await self.getFolderFileDetails(
+      baseFolder, [
+        createOwnCloudFileDetailProperty('fileid', true),
+      ]
+    );
+
+    const detail = folderFileDetails
+    .find(detail => detail.name === fileOrFolderName);
+
+    const fileid = detail.extraProperties['fileid'] as number;
+    result = await self.getObjectCreator(fileid);
+  } catch {}
```

Comment:
Why are you swallowing errors here? Couldn't you just swallow the erro you expect?

Reply:
This was modified like this after a discussion with @kwisatz. As there are several cases (wrong folder, file doesn't exist in folder, issue with activities, etc) it was decided we would swallow the errors, as the end result would have been the same: a reject "Unable to find creator".

Reply:
True, but as you can see above, I imagined it a little different ;) Also what @floriansimon1 is asking is whether you can't try to catch just specific types or errors and let other errors that we don't expect still throw. (I think?!)